src/githubAuth/route.js

```diff
@@ -0,0 +1,77 @@
+import 'dotenv/config';
+import express from 'express';
+import querystring from 'querystring';
+import request from 'request';
+
+import config from './data';
+
+import { 
+  generateRandomState, 
+  getAuthBaseURL,
+  getClientURL
+} from './util';
+
+const router = express.Router();
+const stateKey = 'github-auth-state';
+
+router.get('/login', (req, res) => {
+  console.log('/login');
+  const state = generateRandomState(16);
+  res.cookie(stateKey, state);
+  
+  const queryString = querystring.stringify({
+    client_id: config.github.clientId,
+    redirect_uri: config.github.redirectUrl,
+    state
+  });
+  
+  const url = getAuthBaseURL('/authorize?' + queryString);
+  res.redirect(url);
+});
+
+router.get('/callback', (req, res) => {
+  console.log('/callback');
+  const storedState = req.cookies ? req.cookies[stateKey] : null;
+  const { state } = req.query;
+  
+  if (state && storedState === state) {
+    requestAccessToken(req, res);
+  } else {
+    const queryString = querystring.stringify({ error: 'state_mismatch' });
+    const url = getClientURL(config.client.errorPath + '?' + queryString);
```

Comment:
Hummm, you could use template strings of es6
```suggestion
    const url = getClientURL(`${config.client.errorPath}?${queryString}`);
```